### PR TITLE
Add support for MDX3 and Prettier v3

### LINKED ISSUE #13: Support for Docusaurus v3
I recently tried upgrading some Docusaurus sites I maintain to v3, and it seems like `remark-typescript-tools` doesn't work so well with Docusaurus v3 (or maybe MDX v3?). Here are some screenshots to illustrate what happens:

---

The `import` statements appear as plain text:

Screenshot 2023-11-16 at 12 53 06

---

So do the actual `Tabs` and `TabItem` components:

Screenshot 2023-11-16 at 12 54 00

---

Just checking if you're planning on supporting Docusarus v3? Thanks!

### PR DESCRIPTION
**Pull request by aryaemami59**
## **This PR**:

- [X] Adds support for MDX3.
- [X] Adds support for Prettier v3.
- [X] Updates tooling and dependencies.
- [X] Copies changes from #14.
- [X] Resolves #13.

### CODE REVIEW

**Comment by phryneas** [comment]
Thanks for maintaining the PR - I promise I'll try to get to it soon :)

**Comment by aryaemami59** [comment]
@phryneas Not a problem at all. Thank you and no rush.

File: package.json [hunk 1]
```diff
@@ -11,16 +25,13 @@
...
   },
   "scripts": {
-    "start": "tsdx watch",
-    "build": "tsdx build",
-    "test": "tsdx test",
-    "lint": "tsdx lint",
-    "prepare": "tsdx build"
+    "start": "tsup --watch",
+    "build": "tsup",
+    "test": "true",
```
Comment (phryneas) [inline package.json:30]: I'm fine with dropping the linting steps, but could we keep the tests please? :)

Comment (phryneas) [reply package.json:36]: Oh dang, that was me in the last PR... forget about it 🤦

**Comment by phryneas** [comment]
Ah well, at least one of the tests is running again. That's enough for me right now. I'll get this merged in.

Thank you for taking the time to get this up and running!

**Comment by phryneas** [comment]
I've published this as https://www.npmjs.com/package/remark-typescript-tools/v/2.0.0-alpha.0 - could you test this where you wanted to use it and give me feedback if everything is alright there? Then I'll do the 2.0 final.

**Comment by aryaemami59** [comment]
Yes sirr, I had already tested it (it worked) but I'll test it again just to be sure and let you know you the results shorlty.

**Comment by phryneas** [comment]
Oh I believe you but this stuff always can be slightly different when actually packaged for npm, so double-checking that before a 2.0 just seems like a good idea :)

**Comment by aryaemami59** [comment]
@phryneas Absolutely, I'm going to test it some more to be sure.

**Pull request merged**